src/libraries/System.IO.Compression/tests/ZipArchive/zip_ReadTests.cs

```diff
@@ -61,7 +61,6 @@ public static async Task TestPartialReads(string zipFile, string zipFolder)
                 Stream clamped = new ClampedReadStream(stream, readSizeLimit: 1);
 
                 IsZipSameAsDir(clamped, zfolder(zipFolder), ZipArchiveMode.Read, requireExplicit: true, checkTimes: true);
-                Assert.False(clamped.CanRead, "Clamped stream should be closed at this point"); //check that it was closed
```

Comment:
I was thinking about this and since you're using your own custom clamped stream, You could override Dispose so that it sets the value of a public property called IsDisposed (besides also doing the usual disposing tasks) and then in this test you can check the value of that property to see if we disposed it. But it's ok as it is currently. Just a note for the future.